### ISSUE TITLE
Error from the exercise: javascripting scope

### ISSUE THREAD

**Issue by KadeerCanada** [nodeschool/discussions#2226]
Hello,

I tested last exercise (**Scope**) of `javascripting`. There is no javascript error, but the result is different. I didn't understand what is the problem.

* node version: 8.9.0
* npm version:  5.5.1
* os: macOS High Sierra (v10.13.1)

### Error output
```
 javascripting verify scope.js 

# JAVASCRIPTING

## SCOPE (Exercise 19 of 19)

─────────────────────────────────────────────────────────────────────────────

# O-oh, something isn't working.

# But don't panic!

─────────────────────────────────────────────────────────────────────────────

## Check your solution:

 Solution ===================

 a: 1, b: 8, c: 6

 Your Attempt ===================

 a: 1, b: 2, c: 3

 Difference ===================

 a: 1, b: 82, c: 63

```

### This is my code paste form the exercise `scope.js`: 
```js
 var a = 1, b = 2, c = 3;

(function firstFunction(){
    var b = 5, c = 6;

    (function secondFunction(){
        var b = 8;

        (function thirdFunction(){
            var a = 7, c = 9;

            (function fourthFunction(){
                var a = 1, c = 8;

            })();
        })();
    })();
})();

console.log("a: "+a+", b: "+b+", c: "+c);
```

For testing I change the code like this 
```js
var a = 1, b = 2, c = 3;

(function firstFunction(){
    var b = 5, c = 6;
    console.log("firstFunction ==> a: "+a+", b: "+b+", c: "+c);
    (function secondFunction(){
        var b = 8;
        console.log("secondFunction ==> a: "+a+", b: "+b+", c: "+c);
        (function thirdFunction(){
            var a = 7, c = 9;
            console.log("thirdFunction ==> a: "+a+", b: "+b+", c: "+c);
            (function fourthFunction(){
                var a = 1, c = 8;
                console.log("fourthFunction ==> a: "+a+", b: "+b+", c: "+c);
            })();
        })();
    })();
})();

console.log("a: "+a+", b: "+b+", c: "+c);
```

The result is:
```
firstFunction ==> a: 1, b: 5, c: 6
secondFunction ==> a: 1, b: 8, c: 6
thirdFunction ==> a: 7, b: 8, c: 9
fourthFunction ==> a: 1, b: 8, c: 8
a: 1, b: 2, c: 3
```

I think the `local` variable declared inside the function, didn't work outside of the function.  Is there anything wrong?

**Comment by huberv** [comment]
Hi @KadeerCanada, the variable declarations inside the function blocks create variables which are local to the function bodies.

For example, after 

```javascript
var a = 1, b = 2, c = 3;

(function firstFunction(){
    var b = 5, c = 6;
    // let's assume we stop the execution here
```

we have the following 5 variables:
* a global a with value 1
* a global b with value 2
* a global c with value 3
* a local b with value 5
* a local c with value 6

The local variables with the same name as a global one "shadow" the global variables.

The very last log statement (outside the function bodies) therefore prints the original global variables' values. 

I hope this helps.

Kind regards,
Valentin

**Comment by KadeerCanada** [comment]
Thanks @huberv , 

I fully understand your explanation, and my understanding is the same. 

I checked again from start to finish. The answer of the exercise should be `a: 1, b: 8, c: 6`, but my result is `a: 1, b: 2, c: 3`. The problem is not in the answer.  It is from the question. I didn't understand the question, so my result is different, and my answer can not pass.

I have solved the problem now, and thank you again.

Kadeer
